samples/fpga/gemm_systolic_vectorized.py

```diff
@@ -104,332 +106,605 @@ def make_read_A(state):
     sdfg.add_stream("A_pipe",
-                    dace.float32,
+                    vtype.base_type,
                     transient=True,
-                    shape=(P + 1, ),
+                    shape=("P + 1", ),
                     storage=dace.dtypes.StorageType.FPGA_Local,
                     buffer_size="P")
     sdfg.add_stream("B_pipe",
-                    vec_type,
+                    vtype,
                     transient=True,
-                    shape=(P + 1, ),
+                    shape=("P + 1", ),
+                    buffer_size=MINIMUM_CHANNEL_DEPTH,
                     storage=dace.dtypes.StorageType.FPGA_Local)
+    sdfg.add_stream("B_to_feeder",
+                    vtype,
+                    transient=True,
+                    buffer_size=MINIMUM_CHANNEL_DEPTH,
+                    storage=dace.StorageType.FPGA_Local)
     sdfg.add_stream("C_pipe",
-                    vec_type,
+                    vtype,
                     transient=True,
-                    shape=(P + 1, ),
+                    shape=("P + 1", ),
+                    buffer_size=MINIMUM_CHANNEL_DEPTH,
                     storage=dace.dtypes.StorageType.FPGA_Local)
 
-    make_read_A(state)
-    make_read_B(state, sdfg, vec_width)
-    make_compute(sdfg, state, vec_width)
-    make_write_C(state, sdfg, vec_width)
+    make_read_A(sdfg, state, vtype)
+    make_read_B(sdfg, state, vtype)
+    make_feed_B(sdfg, state, vtype)
+    make_compute(sdfg, state, vtype)
+    make_write_C(sdfg, state, vtype)
 
     return state
 
 
-def make_sdfg(name, vec_width):
+def make_sdfg(name, vtype):
 
     sdfg = dace.SDFG(name)
 
-    pre_state = make_copy_to_fpga_state(sdfg, vec_width)
-    compute_state = make_fpga_state(sdfg, vec_width)
-    post_state = make_copy_to_host_state(sdfg, vec_width)
+    pre_state = make_copy_to_fpga_state(sdfg, vtype)
+    compute_state = make_fpga_state(sdfg, vtype)
+    post_state = make_copy_to_host_state(sdfg, vtype)
 
     sdfg.add_edge(pre_state, compute_state, dace.sdfg.InterstateEdge())
     sdfg.add_edge(compute_state, post_state, dace.sdfg.InterstateEdge())
+
+    if vtype.bytes < 64:
+        sdfg.expand_library_nodes()
+        assert sdfg.apply_transformations_repeated(InlineSDFG) == 2
+
     return sdfg
 
 
 @click.command()
-@click.argument("size_n", type=int)
-@click.argument("size_k", type=int)
-@click.argument("size_m", type=int)
-@click.argument("num-processing-elements", type=int)
+@click.argument("N", type=int)
+@click.argument("K", type=int)
+@click.argument("M", type=int)
+@click.argument("num-pes", type=int)
 @click.argument("vector-width", type=int)
+@click.option("--dtype", default="float32")
+@click.option("--tile-size-n",
+              type=int,
+              default=None,
+              help="Must be a multiple of the number of processing elements.")
+@click.option("--tile-size-m",
+              type=int,
+              default=None,
+              help="Must be a multiple of the vector size.")
```

Comment:
Should it also divide M?

Reply:
Added to both the CLI help and to the overview comment at the top of the file